packages/viewer/src/controllerModel.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable import/no-unresolved */
+import * as THREE from './three/build/three.module.js';
+import { GLTFLoader } from './three/examples/jsm/loaders/GLTFLoader.js';
+import { Constants } from './motion-controllers.module.js';
+/* eslint-enable */
+
+import AssetError from './assetError.js';
+
+const gltfLoader = new GLTFLoader();
+
+class ControllerModel extends THREE.Object3D {
+  constructor() {
+    super();
+    this.xrInputSource = null;
+    this.rootNode = null;
+    this.nodes = [];
+    this.motionController = null;
+    this.loaded = false;
+  }
+
+  async initialize(motionController) {
+    this.motionController = motionController;
+    this.xrInputSource = this.motionController.xrInputSource;
+
+    // Fetch the assets and generate threejs objects for it
+    this.asset = await new Promise(((resolve, reject) => {
+      gltfLoader.load(
+        motionController.assetUrl,
+        (loadedAsset) => { resolve(loadedAsset); },
+        null,
+        () => { reject(new AssetError(`Asset ${motionController.assetUrl} missing or malformed.`)); }
+      );
+    }));
+
+    this.rootNode = this.asset.scene;
+    this.addTouchDots();
+    this.findNodes();
+    this.add(this.rootNode);
+    this.loaded = true;
+  }
+
```

Comment:
Not a problem, just curious: Did you determine this didn't need a `clear()` any more?